src/library_pthread.js

```diff
@@ -178,6 +178,24 @@ var LibraryPThread = {
       if (ENVIRONMENT_IS_PTHREAD && _pthread_self()) ___pthread_tsd_run_dtors();
     },
 
+    onThreadExit: function(tb, exitCode) {
+      // Disable all cancellation so that executing the cleanup handlers won't trigger another JS
```

Comment:
Hmm, the name of this function is a bit odd in the presence of both `threadExit` and `onThreadExit` functions. Maybe something like `deinitThreadPtr`, `shutdownThread` or `runExitHandlersAndDeinitThread`?

Reply:
I like the name `runExitHandlersAndDeinitThread`, changed in 96160f94b